Source/Core/Core/IOS/FS/FileSystemProxy.cpp

```diff
@@ -105,59 +112,78 @@ static IPCReply GetReplyForSuperblockOperation(ResultCode result)
 
 std::optional<IPCReply> FSDevice::Open(const OpenRequest& request)
 {
+  return MakeIPCReply([&](Ticks t) {
+    return Open(request.uid, request.gid, request.path, static_cast<Mode>(request.flags & 3),
+                request.fd, t);
+  });
+}
+
+s64 FSDevice::Open(FS::Uid uid, FS::Gid gid, const std::string& path, FS::Mode mode,
+                   std::optional<u32> ipc_fd, Ticks ticks)
+{
+  ticks.AddTimeBaseTicks(GetIPCOverheadTicks());
+
   if (m_fd_map.size() >= 16)
-    return GetFSReply(ConvertResult(ResultCode::NoFreeHandle));
+    return ConvertResult(ResultCode::NoFreeHandle);
 
-  if (request.path.size() >= 64)
-    return GetFSReply(ConvertResult(ResultCode::Invalid));
+  if (path.size() >= 64)
+    return ConvertResult(ResultCode::Invalid);
 
-  if (request.path == "/dev/fs")
+  const u64 fd = ipc_fd.has_value() ? u64(*ipc_fd) : m_next_fd++;
+
+  if (path == "/dev/fs")
   {
-    m_fd_map[request.fd] = {request.gid, request.uid, INVALID_FD};
-    return GetFSReply(IPC_SUCCESS);
+    m_fd_map[fd] = {gid, uid, INVALID_FD};
+    return fd;
   }
 
-  const u64 ticks = EstimateFileLookupTicks(request.path, FileLookupMode::Normal);
+  ticks.AddTimeBaseTicks(EstimateFileLookupTicks(path, FileLookupMode::Normal));
 
-  auto backend_fd = m_ios.GetFS()->OpenFile(request.uid, request.gid, request.path,
-                                            static_cast<Mode>(request.flags & 3));
-  LogResult(backend_fd, "OpenFile({})", request.path);
+  auto backend_fd = m_ios.GetFS()->OpenFile(uid, gid, path, mode);
```

Comment:
`ticks` doesn't seem to be used.  Did you mean to pass it into `OpenFile` here?

Reply:
Actually, I see this quite a bit.  What is the point of updating the ticks when they are passed in as a copy?

Reply:
The number of ticks is estimated and `ticks` is updated two lines above -- the underlying backend doesn't know anything about timings.

`ticks` isn't an integer, it's a small wrapper around a `u64*`, which is why it is passed by value.

Reply:
Thanks @leoetlino for your explanation.  I believe in my second comment I was confused ^^. In my first comment,  I was aware of the `u64*` but thought the ticks were used in an underlying call somewhere.

Re-reading, I missed an important part.  The ticks are ultimately passed in to the `IPCReply` structure via the `MakeIPCReply` function which I think helps determine the delay call for the command?  Either the delay time is built up from 0 or it is built up from `IPCOverheadTicks` depending on the call.

Sorry for all the dumb comments!

Reply:
Yes, that's correct. MakeIPCReply is used to help construct an `IPCReply` with timing information and without having to duplicate the `IPCReply{..., ticks}` for every return.